src/main/integrations/last-fm/index.ts

```diff
@@ -295,9 +300,13 @@ export default class LastFM implements IIntegration {
 
   private async saveSettings(): Promise<void> {
     try {
-      this.store.set("lastfm.sessionKey", safeStorage.encryptString(this.lastfmDetails.sessionKey).toString("hex"));
-      this.store.set("lastfm.token", safeStorage.encryptString(this.lastfmDetails.token).toString("hex"));
-    } catch {
+      if (this.lastfmDetails.sessionKey) {
+        this.store.set("lastfm.sessionKey", safeStorage.encryptString(this.lastfmDetails.sessionKey).toString("hex"));
+      }
```

Comment:
This was the source of the problem. `sessionKey` was null and throwing a type error and not updating the token in the store. 
We don't have a session key, so we can't get a session key => infinite loop

Reply:
Still need to look into this, but this looks more sound than a try + catch. (this though that's still worth including)

Reply:
the try catch is still there, just does a check so one erroring doesnt break the other from updating

Reply:
I saw the Try + Catch is still there but checking that this isn't breaking anything. But it's all good